### PR TITLE
feat(chore): remove support for node v14,v12,v10

### LINKED ISSUE #110: Remove support for node v14
**Is your feature request related to a problem? Please describe.**
Node v14 reaching its end of life this month. Loopback [removes support for node v14](https://github.com/loopbackio/loopback-next/commit/5425762f1353869994acf081bcda4816e6a9c3b0) in all of its packages we depend on.

Sourceloop packages/services also currently have v12 and v14 as the supported versions.

**Describe the solution you'd like**
Remove the support for node v14 and v12. And add the support for the latest LTS version v18.

**Describe alternatives you've considered**
__

**Additional context**
__

### PR DESCRIPTION
**Pull request by RaghavaroraSF**
## Description

- Version 14, v12, v10 of Nodejs reached its end of life

- BREAKING CHANGE:
End of life of node v14, node v12 and node v10

GH-110

Fixes #110

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by yeshamavani — APPROVED** [review]

**Comment by yeshamavani** [comment]
in the repo there as .d.ts and .js index files present 
they are redundant, please remove those files

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   ],
   "main": "index.js",
   "engines": {
-    "node": ">=8.9"
```
Comment (shubhamp-sf) [inline package.json:11]: Removing LTS version support should be marked as BREAKING CHANGE.
Refer loopback-next's recent node version commits for more insights.

**Review by yeshamavani — APPROVED** [review]

**Review by yeshamavani — CHANGES_REQUESTED** [review]
Why is this marked as feature, no new feature is being introduced

**Comment by yeshamavani** [comment]
After referring loopback decided to go in with feature type

**Review by yeshamavani — APPROVED** [review]

**Pull request merged**